Platformio/HAL/Targets/ESP32/HardwareRevX.cpp

```diff
@@ -71,9 +71,9 @@ void HardwareRevX::init() {
   // Make sure ESP32 is running at full speed
   setCpuFrequencyMhz(240);
 
-  mDisplay = Display::getInstance(std::shared_ptr<HardwareAbstract>(this));
-  mBattery = std::make_shared<Battery>(ADC_BAT,CRG_STAT);
-  mWifiHandler = wifiHandler::getInstance(std::shared_ptr<HardwareAbstract>(this));
+  this->mDisplay = Display::getInstance(std::shared_ptr<HardwareAbstract>(this));
+  this->mBattery = std::make_shared<Battery>(ADC_BAT,CRG_STAT);
+  this->mWifiHandler = wifiHandler::getInstance(std::shared_ptr<HardwareAbstract>(this));
```

Comment:
do the HW components still need a ref to the abstract ? This is the cyclic dependency I was referring to.

Reply:
I think I overlooked this. No we can remove the HardwareAbstract for wifi. However, I'm not sure we can in general be sure that this is not needed. If at some point we need hardware to communicate with each other it might be needed.